pkg/router/controller/route_validator.go

```diff
@@ -0,0 +1,144 @@
+package controller
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/golang/glog"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util/sets"
+	"k8s.io/kubernetes/pkg/watch"
+
+	"github.com/openshift/origin/pkg/cmd/util/variable"
+	routeapi "github.com/openshift/origin/pkg/route/api"
+	"github.com/openshift/origin/pkg/router"
+)
+
+// RouteHostFunc returns a host for a route. It may return an empty string.
+type RouteHostFunc func(*routeapi.Route) string
+
+// HostForRoute returns the host set on the route.
+func HostForRoute(route *routeapi.Route) string {
+	return route.Spec.Host
+}
+
+// routeSelectionFunc returns a function that returns the host for a route,
+// possibly generated according to the specified template.
+func routeSelectionFunc(hostnameTemplate string,
+	overrideHostname bool, overrideExceptions sets.String) RouteHostFunc {
+	if len(hostnameTemplate) == 0 {
+		return HostForRoute
+	}
+	return func(route *routeapi.Route) string {
+		if (!overrideHostname ||
+			overrideExceptions.Has(route.ObjectMeta.Namespace)) &&
+			len(route.Spec.Host) > 0 {
+			return route.Spec.Host
+		}
+		s, err := variable.ExpandStrict(hostnameTemplate,
+			func(key string) (string, bool) {
+				switch key {
+				case "name":
+					return route.Name, true
+				case "namespace":
+					return route.Namespace, true
+				default:
+					return "", false
+				}
+			})
+		if err != nil {
+			return ""
+		}
+		return strings.Trim(s, "\"'")
+	}
+}
+
+// RouteValidator implements the router.Plugin interface and validates
+// routes before passing them along to another router.Plugin.
+type RouteValidator struct {
+	plugin router.Plugin
+
+	recorder RejectionRecorder
```

Comment:
Why do we need an entire new object for just logging? Are we expected to persist this information ever?